p2p/src/main/kotlin/maru/p2p/MaruPeer.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.p2p
+
+import java.util.Optional
+import maru.p2p.messages.Status
+import maru.p2p.messages.StatusMessageFactory
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import tech.pegasys.teku.networking.p2p.network.PeerAddress
+import tech.pegasys.teku.networking.p2p.peer.DisconnectReason
+import tech.pegasys.teku.networking.p2p.peer.DisconnectRequestHandler
+import tech.pegasys.teku.networking.p2p.peer.Peer
+import tech.pegasys.teku.networking.p2p.peer.PeerDisconnectedSubscriber
+import tech.pegasys.teku.networking.p2p.reputation.ReputationAdjustment
+import tech.pegasys.teku.networking.p2p.rpc.RpcMethod
+import tech.pegasys.teku.networking.p2p.rpc.RpcRequestHandler
+import tech.pegasys.teku.networking.p2p.rpc.RpcResponseHandler
+import tech.pegasys.teku.networking.p2p.rpc.RpcStreamController
+
+interface MaruPeer : Peer {
+  fun getStatus(): Status?
+
+  fun sendStatus(): SafeFuture<Status>
```

Comment:
Should we update status on every block? Or when remote peer's block height is close to Head, we optimistically assume it's actually synced?

Reply:
Maybe in addition to block heights we can add a field whether peer thinks it's synced

Reply:
As discussed on the sync call, it's not clear whether we will need this or not. If we do we can this in later